app/renderer/js/utils/config-util.js

```diff
@@ -66,7 +66,24 @@ class ConfigUtil {
 	}
 
 	setConfigItem(key, value) {
-		this.db.push(`/${key}`, value, true);
+		const item = {};
+		if (this.isConfigItemExists(key) && this.getConfigItem(key).admin) {
+			// if admin is set to true, user is not allowed to change that setting
+			return;
+		}
+		if (this.isConfigItemExists(key) && (value === null || value.admin === undefined)) {
```

Comment:
All of these conditionals look fragile can you explain what these are if statement are doing for all the cases. I definitely feel there is cleaner and easier way to write these here. Also, I recommend checking `value.admin !== true` here.